src/core/mac/mac_filter.cpp

```diff
@@ -142,109 +136,112 @@ otError Filter::RemoveAddress(const ExtAddress &aExtAddress)
 
 void Filter::ClearAddresses(void)
 {
-    for (uint8_t i = 0; i < GetMaxEntries(); i++)
+    FilterEntry *entry = &mFilterEntries[0];
+
+    for (size_t num = OT_ARRAY_LENGTH(mFilterEntries); num != 0; num--, entry++)
```

Comment:
Here may use the pointer check as well?

Reply:
Thanks. I missed these two. I changed them in new pushed commit.